coala_quickstart/interaction/Logo.py

```diff
@@ -0,0 +1,58 @@
+from coala_quickstart.Strings import coala_bear_logo, welcome_messages
+
+
+def text_wrap(*args, delimiter=' ', end='\n', limit=80):
+    """
+    Breaks it word wise when going further than ``limit`` chars.
+
+    :param args:
+        The stuff you want to have printed.
+    :param delimiter:
+        Will be placed between all the args.
+    :param limit:
+        Char limit.
+    :param end:
+        An ending, will be put as last character and doesn't count into the
+        char limit.
+    """
+    output = delimiter.join(args)
+    lines = output.splitlines(keepends=True)
+    results = [""]
+    for line in lines:
+        curr_print = line
+        while len(curr_print.rstrip('\n')) > limit:
+            splitpos = curr_print[:limit].rfind(' ')
+            if splitpos < 0:
+                # Word too long, search for a space left from limit at least
+                splitpos = curr_print.find(' ')
+                if splitpos < 0:
+                    break  # Break out and print the long thing in 28
+
+            results.append(curr_print[:splitpos])
+            curr_print = curr_print[splitpos+1:]
+
+        results.append(curr_print)
+
+    return results
+
+
+def print_welcome_message(printer):
+    """
+    Prints the coala bear logo alongside a welcome message.
+
+    :param printer:
+        A ``ConsolePrinter`` object used for console interaction.
+    """
+    max_length = 80 - len(max(coala_bear_logo, key=len))
```

Comment:
Ooh you're putting them side by side!?!
On 22 Jun 2016 12:35, "Adhityaa Chandrasekar" <notifications@github.com>
wrote:

> In coala_quickstart/interaction/Logo.py
> <https://github.com/coala-analyzer/coala-quickstart/pull/20#discussion_r68030836>
> :
>
> > +            results.append(curr_print[:splitpos])
> > +            curr_print = curr_print[splitpos+1:]
> > +
> > +        results.append(curr_print)
> > +
> > +    return results
> > +
> > +
> > +def print_welcome_message(printer):
> > +    """
> > +    Prints the coala bear logo alongside a welcome message.
> > +
> > +    :param printer:
> > +        A ``ConsolePrinter`` object used for console interaction.
> > +    """
> > +    max_length = 80 - len(max(coala_bear_logo, key=len))
>
> @sils1297 <https://github.com/sils1297> what's the mistake?
>
> BTW I'm deleting the other four messages, they're cluttering the comments
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/coala-analyzer/coala-quickstart/pull/20/files/40e99e394aa8e527aa4b3cfbf035694225d6d9f2..8efb8ca566ad20e07e1d580b880c7456b56779e4#r68030836>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AFc6KF77f4L7dlqUQlVIORgt-RNqTy4vks5qORALgaJpZM4IzlPV>
> .
>